deps/src/singular.cpp

```diff
@@ -5,6 +5,25 @@
 #include "ideals.h"
 #include "matrices.h"
 
+std::string singular_return;
+std::string singular_error;
+std::string singular_warning;
+extern int  inerror;
+
+void WerrorS_for_julia(const char * s)
+{
+    singular_error += s;
+}
```

Comment:
This code is not thread-safe, but I assume neither is any other part of Singular / Singular.jl, so probably doesn't matter?

Reply:
Non of this is thread-safe, no. I would not really think that this is necessary here for now.